code/ZAS/Turf.dm

```diff
@@ -5,18 +5,19 @@
 /turf/var/datum/gas_mixture/air
 
 /turf/simulated/proc/update_graphic(list/graphic_add = null, list/graphic_remove = null)
-	if(LAZYLEN(graphic_add))
-		add_overlay(graphic_add, priority = TRUE)
+	if (LAZYLEN(graphic_add))
+		vis_contents += graphic_add
 	if(LAZYLEN(graphic_remove))
-		cut_overlay(graphic_remove, priority = TRUE)
+		vis_contents -= graphic_remove
```

Comment:
todo: investigate if this works

Reply:
seems to work. this handles overlays such as plasma gas and fire. cant be 100% sure without live testing